src/main/java/uk/gov/pay/adminusers/persistence/entity/ServiceEntity.java

```diff
@@ -157,18 +158,25 @@ public static ServiceEntity from(Service service) {
         return serviceEntity;
     }
 
-    public void addServiceName(ServiceNameEntity serviceName) {
+    public void addOrUpdateServiceName(ServiceNameEntity serviceName) {
         serviceName.setService(this);
-        this.serviceName.add(serviceName);
+        final Optional<ServiceNameEntity> maybeNameEntity = this.serviceNames.stream()
+                .filter(s -> s.equals(serviceName))
```

Comment:
I cannot have `name` into the equals method because when `ServiceEntity` will add to its set of `ServiceNameEntity` then it will fail the equality check and we might end-up having two records for the same language, instead of updating the language with the second value.